### ISSUE TITLE
Add a way to run different commands with/without count

### ISSUE THREAD

**Issue by The-Compiler** [qutebrowser/qutebrowser#7213]
> @The-Compiler I just upgraded to the newest version, and saw that `tab-focus` without arguments is deprecated (since the version mentioned in this thread). What should be used now instead to use qutebrowser tabs just like Vim tabs?
>
> I mapped `gt` to `tab-focus` and `gT` to `tab-prev`, allowing me to go to a specific tab via `<n>gt` and going to the next tab without a count. How can I do this now?

_Originally posted by @leo848 in https://github.com/qutebrowser/qutebrowser/discussions/6390#discussioncomment-2821823_

---

We could just un-deprecate this behavior of `tab-focus` without arguments (see ff63d0c1b0331a8fee8808dbb536a0ebbe3bbb68 and #6314 for some context)...

However, I wonder if it would make sense to introduce something more generic which wouldn't require writing a userscripts for those things. For example, something like `cmd-if-count tab-focus tab-next` which runs `tab-focus` if there was a count, and `tab-next` if not.

cc @mschilli87 @toofar from the original discussion.

**Comment by leo848** [comment]
@The-Compiler Good idea, however I feel like this should be more generalized. Maybe it could be like `cmd-if <option> <cmd-if-true> <cmd-if-false>` where option could be from a set of values (like `count, history-back, history-forwards, tab-mute` etc.). 

Also `cmd-if-true` and `cmd-if-false` need to be separated somehow. Of course having to use quotes (`"`/`'`) would be an option, but it doesn't look that nice if the config file if many commands are covered in wrapped strings.

**Comment by The-Compiler** [comment]
I don't think it makes sense to generalize this further. For more complex things, I'd rather have people write an extension (#30) or a userscript rather than trying to shoehorn everything into a `cmd-if` command.

As for the separation, that would already work out of the box (and use quotes indeed) when using two arguments to `cmd-if-count`.

**Comment by ironwod** [comment]
@The-Compiler I want to make sure, that I understand this issue correctly. The `tab-focus` command without count argument should stay deprecated and new command (`cmd-if-count`) should be created to work as a "wrapper" around `tab-focus` and `tab-next` commands.

Is that right? If so, I will take that.

**Comment by The-Compiler** [comment]
Sounds good to me!

I wonder if we should also deprecate `:cmd-set-text --run-with-count` at this point, as e.g. `set-cmd-text -sr :tab-focus` (from the default bindings) can be expressed as `:cmd-if-count tab-focus "set-cmd-text -s tab-focus"` then. Somewhat undecided though, as it's more verbose.
